azurelinuxagent/common/conf.py

```diff
@@ -228,6 +228,11 @@ def get_switch_default_value(option):
     raise ValueError("{0} is not a valid configuration parameter.".format(option))
 
 
+def get_switch_no_default_value(key, conf=__conf__):
+    # If specified, return the value else return None
+    return conf.get_switch(key, None)
+
```

Comment:
this function feels a little weird, doesn't it? 

Maybe

```
    def is_present(self, key):
        """
        Returns True if the given flag is present in the config file.
        """
        return self.values.get(key) is not None

```

Reply:
updated